packages/@n8n/ai-workflow-builder.ee/src/tools/builder-tools.ts

```diff
@@ -29,6 +31,8 @@ export function getBuilderTools({
 	instanceUrl?: string;
 }): BuilderTool[] {
 	return [
+		createCategorizePromptTool(llmComplexTask, logger),
+		createGetBestPracticesTool(),
```

Comment:
I wonder if we really need `categorizePromptTool` 🤔 Wouldn't the root agent already have enough context to determine the techniques it should request best practices for?

Reply:
The benefit of having it as a tool/separate chain means that is re-usable/testable outside of the root agent. When we break down into sub-agents I think the discovery subagent could look after it entirely, but if we insert this logic into the root agent it means we can't use it in scripts/separate it out for individual integration testing.